nmdc_runtime/site/export/ncbi_xml_utils.py

```diff
@@ -96,6 +96,38 @@ def fetch_omics_processing_from_biosamples(all_docs_collection, biosamples_list)
     return biosample_data_objects
 
 
+def fetch_library_preparation_from_biosamples(all_docs_collection, biosamples_list):
+    biosample_lib_prep = []
+
+    for biosample in biosamples_list:
+        biosample_id = biosample["id"]
+
+        # Step 1: Find any document with biosample id as has_input
+        initial_query = {"has_input": biosample_id}
+        initial_document = all_docs_collection.find_one(initial_query)
+
+        if not initial_document:
+            continue
+
+        initial_output = initial_document.get("has_output")
+        if not initial_output:
+            continue
+
+        # Step 2: Use has_output to find the library preparation document
+        for output_id in initial_output:
+            lib_prep_query = {
+                "has_input": output_id,
+                "designated_class": "nmdc:LibraryPreparation",
```

Comment:
change designated_class to `type`

Reply:
I created issue https://github.com/microbiomedata/nmdc-runtime/issues/630 about this.